atomata/capstone_WebClient
# Overhaul of Props Drilling into Global Context Hooks

### Summary 
Refactoring of Certain Hooks into Global Context Hooks in order to avoid Props drilling
### Details
This PR adds Global Context Hooks in a high level component (namely experience.tsx), and the following hooks and their sub-hooks have been refactored to be used as Global Context Hooks
1. experienceData
2. actionList
3. userId

### How to test?
1. Spin up the app and make sure the editor successfully passes data back and forth. 
2. The app should work exactly like the current main branch.

### Checks
